addons/website_event_booth_sale/models/sale_order.py

```diff
@@ -29,27 +29,38 @@ def _website_product_id_change(self, order_id, product_id, qty=0, **kwargs):
             order_line = self.env['sale.order.line'].sudo().search([
                 ('id', 'in', self.order_line.ids),
                 ('event_booth_pending_ids', 'in', event_booth_pending_ids)])
-            booths = self.env['event.booth'].browse(event_booth_pending_ids)
-            new_registration_ids = [Command.create({
-                                        'event_booth_id': booth.id,
-                                        **kwargs.get('registration_values'),
-                                    }) for booth in booths]
-            if order_line:
-                event_booth_registration_ids = [Command.delete(reg.id)
-                                                for reg in order_line.event_booth_registration_ids] + new_registration_ids
-            else:
-                event_booth_registration_ids = new_registration_ids
+            booths = self.env['event.booth'].browse(event_booth_pending_ids).with_context(pricelist=self.pricelist_id.id)
+            if order_line.event_booth_pending_ids.ids != event_booth_pending_ids:
+                new_registration_ids = [Command.create({
+                                            'event_booth_id': booth.id,
+                                            **kwargs.get('registration_values'),
+                                        }) for booth in booths]
+                if order_line:
+                    event_booth_registration_ids = [Command.delete(reg.id)
+                                                    for reg in order_line.event_booth_registration_ids] + new_registration_ids
+                else:
+                    event_booth_registration_ids = new_registration_ids
+                values.update(event_booth_registration_ids=event_booth_registration_ids)
 
+            if self.pricelist_id.discount_policy == 'without_discount':
+                price_unit = sum(booth.price for booth in booths)
```

Comment:
By the way, I would assume that "price_unit" is the price for a single unit, is that intended to sum for every booth here?
Maybe I'm misunderstanding that.